File: src/main/java/com/samcox/ranker/UserCredentials.java
package com.samcox.ranker;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public class UserCredentials {
  @NotBlank(message = "User must have username")
  private String username;
  @NotBlank(message = "User must have password")
  @Size(min = 8, max = 64, message = "Password must be between 8 and 64 characters")
  @Pattern(regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@$!%*?&])[A-Za-z\\d@$!%*?&]{8,}$",
    message = "Password must contain at least one uppercase letter, one lowercase letter," +
    " one number and one special character")
  private String password;

  public String getUsername() {
    return username;
  }
  public String getPassword() {
    return password;
  }
  public void setUsername(String username) {
    this.username = username;
  }
  public void setPassword(String password) {
    this.password = password;
  }
}

File: frontend/src/app/user-credentials.ts
export class UserCredentials {

  username: string;
  password: string;

  constructor() {
    this.username = '';
    this.password = '';
  }

  setUsername(username: string) {
    this.username = username;
  }

  setPassword(password: string) {
    this.password = password;
  }

  getUsername() {
    return this.username;
  }

  getPassword() {
    return this.password;
  }

}
